src/main/java/uk/gov/register/views/representations/spreadsheet/RecordSpreadsheet.java

```diff
@@ -0,0 +1,48 @@
+package uk.gov.register.views.representations.spreadsheet;
+
+import org.apache.poi.ss.usermodel.Workbook;
+import uk.gov.register.core.Entry;
+import uk.gov.register.core.Field;
+import uk.gov.register.views.ItemView;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public interface RecordSpreadsheet {
+
+    default void process(final Iterable<Field> fields, final Map<Entry, List<ItemView>> records, final String sheetLabel,
+                         final OutputStream entityStream) throws IOException {
+        final List<String> fieldNames = new ArrayList<>();
+        final List<Map<String, String>> elements = new ArrayList<>();
+        final Workbook workbook;
+
+        fieldNames.add("index-entry-number");
+        fieldNames.add("entry-number");
+        fieldNames.add("entry-timestamp");
+        fieldNames.add("key");
+
+        fields.forEach(field -> fieldNames.add(field.fieldName));
+
+        records.forEach((key, items) -> {
+            final Map<String, String> element = new HashMap<>();
+
+            element.put("index-entry-number", key.getIndexEntryNumber().toString());
+            element.put("entry-number", key.getEntryNumber().toString());
+            element.put("entry-timestamp", key.getTimestampAsISOFormat());
+            element.put("key", key.getKey());
+
+            items.forEach(item -> item.getContent()
```

Comment:
A potential problem here (probably unlikely to occur) would be if we had a property in the item with the same name, i.e. `key`. We could avoid this by namespacing the item content, but this might be a step too far for our needs at the moment.

Reply:
I agree, field names must be unique but, as you have said, for now, application fields have very specific names and I hope there is no going to be conflicts. If we find conflicts  a different approach can be taken.

Reply:
:+1: